pkg/controller/grouppermission/grouppermission_controller_test.go

```diff
@@ -0,0 +1,344 @@
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      operatorconfig.OperatorName,
+			Namespace: operatorconfig.OperatorNamespace,
+		},
+	}
+}
+
+// create fake client to mock API calls
+func newTestReconciler() *ReconcileGroupPermission {
+	return &ReconcileGroupPermission{
+		client: client.NewFakeClient(),
+		scheme: scheme.Scheme,
+	}
+}
+
+func mockClusterRole() *rbacv1.ClusterRole {
+	return &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "dedicated-admins-cluster",
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"rbac.authorization.k8s.io"},
+				Resources: []string{"clusterrolebindings"},
+				Verbs:     []string{"create", "delete", "get", "list"},
+			},
+		},
+	}
+}
+
+func mockClusterRoleBinding() *rbacv1.ClusterRoleBinding {
+	return &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "exampleClusterRoleName" + "-" + "exampleGroupName",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind: "Group",
+				Name: "exampleGroupName",
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind: "ClusterRole",
+			Name: "exampleClusterRoleName",
+		},
+	}
+}
+
+func mockGroupPermissionSpec() *managedv1alpha1.GroupPermission {
+	return &managedv1alpha1.GroupPermission{
+		Spec: managedv1alpha1.GroupPermissionSpec{
+			GroupName: "exampleGroupName",
+			ClusterPermissions: []managedv1alpha1.ClusterPermission{
+				{
+					ClusterRoleName: "exampleClusterRoleName",
+				},
+				{
+					ClusterRoleName: "exampleClusterRoleNameTwo",
+				},
+			},
+		},
+	}
+}
+
+func mockGroupPermissionStatus(time metav1.Time) *managedv1alpha1.GroupPermission {
+	return &managedv1alpha1.GroupPermission{
+		Status: managedv1alpha1.GroupPermissionStatus{
+			Conditions: []managedv1alpha1.Condition{
+				{
+					LastTransitionTime: time,
+					ClusterRoleName:    "exampleClusterRoleName",
+					Message:            "exampleMessage",
+					Status:             true,
+					State:              "exampleState",
+				},
+			},
+		},
+	}
+}
+
+func TestInvalidGroupPermission(t *testing.T) {
+	ctx := context.TODO()
+	reconciler := newTestReconciler()
+	defer reset(ctx, reconciler)
+
+	// Register operator types with the runtime scheme.
+	s := scheme.Scheme
+
+	//Add api to scheme
+	if err := routev1.AddToScheme(s); err != nil {
+		t.Fatalf("Unable to add route scheme: (%v)", err)
+	}
+
+	// creates a groupPermission object
+	nerr := reconciler.client.Create(ctx, makeGroupPermission())
+	if nerr != nil {
+		t.Errorf("Couldn't create required GroupPermission object for test: %s", nerr)
+	}
+
+	// create a request - mock the API call
+	request := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      "test",
+			Namespace: "",
+		},
+	}
+
+	// process the request
+	res, err := reconciler.Reconcile(request)
+	if res.Requeue {
+		t.Error("Didn't expect to requeue")
+	}
+	if err != nil {
+		t.Errorf("Got an unexpected error: %s", err)
+	}
+}
+
+func TestValidGroupPermission(t *testing.T) {
```

Comment:
This test doesn't seem to match the name

Reply:
changing test name to "TestPopulateCrClusterRoleNames"